### PR TITLE
Added @var for IDE-typehinting

### PR DESCRIPTION
**Pull request by edsonmedina**
PHPStorm complains about variable not being a callable

### CODE REVIEW

File: src/Prophecy/Prophecy/MethodProphecy.php [hunk 1]
```diff
@@ -86,6 +86,7 @@ public function __construct(ObjectProphecy $objectProphecy, $methodName, $argume
                     case 'Traversable':
                     case 'Generator':
                         // Remove eval() when minimum version >=5.5
+                        /** @var callable  */
```
Comment (stof) [inline src/Prophecy/Prophecy/MethodProphecy.php]: missing variable name

Comment (edsonmedina) [inline src/Prophecy/Prophecy/MethodProphecy.php]: I'm just being consistent. 
The other places in the code using @var type-hints don't use variable names either.

Comment (edsonmedina) [inline src/Prophecy/Prophecy/MethodProphecy.php]: I can add it if you prefer

File: src/Prophecy/Prophecy/MethodProphecy.php [hunk 2]
```diff
@@ -86,6 +86,7 @@ public function __construct(ObjectProphecy $objectProphecy, $methodName, $argume
                     case 'Traversable':
                     case 'Generator':
                         // Remove eval() when minimum version >=5.5
+                        /** @var callable $generator */
                         $generator = eval('return function () { yield; };');
```
Comment (dawehner) [inline src/Prophecy/Prophecy/MethodProphecy.php]: Is it just me, or is the minimum version 5.5 already, when you use generators?

Comment (aik099) [inline src/Prophecy/Prophecy/MethodProphecy.php]: If you look up, then this code is wrapped with an IF statement where PHP 7.0 or above is required.

**Comment by ciaranmcnulty** [comment]
Why add a @var when it's immediately returned anyhow?

**Comment by jakzal** [comment]
@ciaranmcnulty what's returned is a result of calling the generator, not the generator itself. 

The way PHPStorm  complains looks like this:

phpspec-phpstorm

Not sure if this is a big deal. Endusers won't see it.

**Pull request merged**